kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/IngressReconciler.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.time.Clock;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.javaoperatorsdk.operator.api.config.informer.InformerEventSourceConfiguration;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.ErrorStatusUpdateControl;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
+import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
+
+import io.kroxylicious.kubernetes.api.common.Condition;
+import io.kroxylicious.kubernetes.api.common.ConditionBuilder;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngress;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngressBuilder;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Reconciles a {@link KafkaProxyIngress} by checking whether the {@link KafkaProxy}
+ * referenced by the {@code spec.proxyRef.name} actually exists, setting a
+ * {@link Condition.Type#ResolvedRefs} {@link Condition} accordingly.
+ */
+public class IngressReconciler implements
+        Reconciler<KafkaProxyIngress> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IngressReconciler.class);
+    public static final String PROXY_EVENT_SOURCE_NAME = "proxy";
+    private final Clock clock;
+
+    IngressReconciler(Clock clock) {
```

Comment:
I think we should end up with all the Reconcilers taking a clock for mocking purposes. This is the first step.